### DISCUSSION TITLE
Trying to run UC on a Raspberry Pi

### DISCUSSION THREAD

**Discussion by jvbs013** [ultrafunkamsterdam/undetected-chromedriver#1781] [Q&A]
Getting the following error:
OSError: [Errno8] Exec format error: '/home/jvbs/.local/share/undetected_chromedriver/undetected_chromedriver'

I'm already using '--headless', '--no-sandbox' and 'disable-dev-shm-usage' as ChromeOptions

**Comment by KebabLord** [comment]
You are getting this error because UC executable binary is not working, think of it as trying to run an "exe" on a "linux".

The reason is the "automatic patcher" of UC downloads the driver binary by looking at the operating system, if it's windows, it downloads win32 driver, if it's linux, it downloads linux64, if it's darwin it downloads mac64 driver.

The thing is, Raspberry is running linux, so UC will assume it's a 64bit PC, but actually it has ARM architecture. UC should been refusing to install instead of proceeding with incorrect architecture

https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/0aa5fbe252370b4cb2b95526add445392cad27ba/undetected_chromedriver/patcher.py#L111-L116


Straight to the point, as i know chromewebdriver project doesn't provide binaries for armv7 or arm64, you might need to compile it yourself and update the UC library to work on it. But there is a better workaround, you can just use driverless webdriver projects such as :
https://github.com/ultrafunkamsterdam/nodriver
https://github.com/kaliiiiiiiiii/Selenium-Driverless

If you are not good at programming in async or need gui etc. Mozilla's webdriver "geckodriver" also provides binaries for ARM, AARCH64, ARMv7 etc.
https://github.com/mozilla/geckodriver
